File: packages/sdkwork-utils-go/optional.go
package utils

import (
	"fmt"
	"strings"
	"time"
)

func Coalesce(values ...*string) *string {
	for _, value := range values {
		if value == nil {
			continue
		}
		if !IsBlank(value) {
			trimmed := Trim(*value)
			return &trimmed
		}
	}
	return nil
}

func DefaultIfBlank(value *string, defaultValue string) string {
	if IsBlank(value) {
		return defaultValue
	}
	return Trim(*value)
}

File: packages/sdkwork-utils-typescript/src/optional.ts
import { isBlank } from "./string.js";

export function coalesce(...values: Array<string | null | undefined>): string | undefined {
  for (const value of values) {
    if (!isBlank(value)) {
      return value!.trim();
    }
  }
  return undefined;
}

export function defaultIfBlank(
  value: string | null | undefined,
  defaultValue: string,
): string {
  return isBlank(value) ? defaultValue : value!.trim();
}
